unixfs.go

```diff
@@ -9,9 +9,18 @@ import (
 	proto "github.com/gogo/protobuf/proto"
 
 	dag "github.com/ipfs/go-merkledag"
+
+	ipld "github.com/ipfs/go-ipld-format"
 	pb "github.com/ipfs/go-unixfs/pb"
 )
 
+// A LinkResult for any parallel enumeration of links
+// TODO: Should this live in go-ipld-format?
+type LinkResult struct {
```

Comment:
I'm not really a fan of grouping errors like this since it has no conceptual anchor in the IPFS mental model of the user, but I'm also not sure what would be the most  idiomatic Go solution here. Normally in asynchronous cases like this I see a separate channel for potential errors, e.g., `EnumLinksAsync` instead of returning a `(chan, error)` (where the error is always `nil`) could return a `(chan, chan)` one for links and one for errors.

Reply:
that's an interesting idea and it makes sense. FYI, I'm copying the style used here: https://github.com/ipfs/go-ipfs/pull/5404 -- I think we should standardize whatever we end up with.

Reply:
@magik6k thoughts?

Reply:
Using 2 separate channels tends to be really annoying for channel closing logic as we need to make sure both channel are closed, I'd rather keep 1 channel with a struct.,

Reply:
Yeah, this ("result objects like this") is, unfortunately, idiomatic go. Ideally, we'd have sum-types but we don't.